int32 skill_type
int32 meta_skill_type
int32 meta_skill_id

# Sensor topic to subscribe to
string[] sensor_topics
uint32[] sensor_value_sizes
float32[] initial_sensor_values

# traj gen
int32 traj_gen_type
uint32 num_traj_gen_params
float32[] traj_gen_params

# fbc
int32 feedback_controller_type
uint32 num_feedback_controller_params
float32[] feedback_controller_params

# termination
int32 termination_type
uint32 num_termination_params
float32[] termination_params 

# timer
int32 timer_type
uint32 num_timer_params
float32[] timer_params
---
# Result - happens when the skill terminates
uint32 num_execution_result
float32[] execution_result
---
# Feedback message - happens during skill execution 
uint32 num_execution_feedback
float32[] execution_feedback
